src/common/getters.ts

```diff
@@ -775,6 +442,389 @@ export function getInherit(vc: DatasetWithId): boolean {
   );
 }
 
+const WELL_KNOWN_KEYS = [
+  "forPersonalData",
+  "forPurpose",
+  "isProvidedTo",
+  "isProvidedToController",
+  "isProvidedToPerson",
+  "mode",
+  "inherit",
+  "hasStatus",
+  "isConsentForDataSubject",
+  gc.forPersonalData.value,
+  gc.forPurpose.value,
+  gc.isProvidedTo.value,
+  gc.isProvidedToController.value,
+  gc.isProvidedToPerson.value,
+  gc.hasStatus.value,
+  gc.isConsentForDataSubject.value,
+  acl.mode.value,
+  INHERIT.value,
+] as const;
+
+/**
+ * Reads all the custom fields in the consent section of the provided Access Credential.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customString"),
+ *       value: "custom value",
+ *     },
+ *     {
+ *       key: new URL("https://example.org/ns/customInteger"),
+ *       value: 1,
+ *     },
+ *   ]),
+ * });
+ * const customFields = getCustomFields(accessRequest);
+ * // s is "custom value"
+ * const s = customFields["https://example.org/ns/customString"];
+ * // i is 1
+ * const i = customFields["https://example.org/ns/custominteger"];
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access Request)
+ * @returns an object keyed by the custom fields names, associated to their values.
+ * @since unreleased
+ */
+export function getCustomFields(
+  accessCredential: DatasetWithId,
+): Record<string, unknown> {
+  const credentialObject = JSON.parse(JSON.stringify(accessCredential));
```

Comment:
This uses the underlying JSON serialization of the VC, because there are two issues with the RDF dataset semantics: 
- `{..., "https://some.custom/field": true, ...}` and `{..., "https://some.custom/field": [true], ...}` will both result in the same quad being present in the dataset, so we don't know whether it initially was a scalar or an array.
- `{..., "https://some.custom/field": [true, true, true], ...}` will result in a single quad, because all the objects are the same (this applies to all literal types). This means getting back a value would be confusing, as arrays would be collapsed in a Set-like manner.